File: 240321/체크판위에서 2/on-the-checkboard-2.py
import sys

R, C = map(int,sys.stdin.readline().split())
board = [sys.stdin.readline().strip().split() for _ in range(R)]

color = {'W':'B','B':'W'}
now = board[0][0]

count = 0
for row in range(1,R):
    for col in range(1,C):
        if board[row][col] == color[now]:
            for r in range(row+1,R-1):
                for c in range(col+1,C-1):
                    if board[r][c] == now:
                        count += 1

print(count)

File: 240404/체크판위에서 2/on-the-checkboard-2.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int R, C, count = 0;
    char board[15][15],temp;

    scanf("%d %d",&R,&C);
    int r = 0, c = 0;
    while(r < R && c < C) {
        scanf("%c",&temp);
        if (temp != ' ' && temp != '\n') {
            board[r][c] = temp;
            c++;
            if(c == C) {
                c = 0;
                r++;
            }
        }
    }
    
    if (board[0][0] == board[R-1][C-1])
        count = 0;
    else {
        count = 0;
        for(int row = 1; row < R; row++)
            for(int col = 1; col < C; col++)
                if (board[row][col] != board[0][0])
                    for(int r = row+1; r < R-1; r++)
                        for(int c = col+1; c < C-1; c++)
                            if (board[r][c] == board[0][0])
                                count++;
    }

    printf("%d\n",count);

    return 0;
}